Marlin/Marlin_main.cpp

```diff
@@ -3843,16 +3791,14 @@ inline void gcode_G28() {
   /**
    * G30: Do a single Z probe at the current XY
    */
-  inline void gcode_G30() {
+  inline float gcode_G30() {
 
     setup_for_endstop_or_probe_move();
 
-    deploy_z_probe();
-
-    stepper.synchronize();
-
     // TODO: clear the leveling matrix or the planner will be set incorrectly
-    float measured_z = run_z_probe(); // clears the ABL non-delta matrix only
+    float measured_z = probe_pt(current_position[X_AXIS] + X_PROBE_OFFSET_FROM_EXTRUDER,
```

Comment:
I thought of using `probe_pt`, but I thought it would be a waste program of space adding `probe_pt` if `run_z_probe` can do the job. And `probe_pt` only uses `run_z_probe` anyway. But then, `probe_pt` has become simpler.

I'm curious about why `gcode_G30` should return a `float` now? So far we haven't made any of the gcode handlers callable. But I note that since it takes no parameters, it's a candidate for a general function. But then… it is tempting to think about extending `G30` to take `X` and `Y` parameters, and even a flag `S` indicating whether or not to stow afterward.

I don't want to get too far ahead of Unified Bed Leveling, but I'm holding out for…
```cpp
void gcode_G29() {
  // get flags and options, then...
  prepare_for_probing();
  goto_first_probe_point();
  do { probe_z_here(); probe_report(); } while (goto_next_probe_point());
  finish_probing();
}
```